params/config.go

```diff
@@ -695,6 +695,70 @@ func isMaxCodeSizeConfigCompatible(c1, c2 *ChainConfig, head *big.Int) (error, *
 	return nil, big.NewInt(0), big.NewInt(0)
 }
 
+// Quorum
+//
+// checks if changes to transitions proposed are compatible
+// with already existing genesis data
+func isTransitionsConfigCompatible(c1, c2 *ChainConfig, head *big.Int) (error, *big.Int, *big.Int) {
+	if len(c1.Transitions) == 0 && len(c2.Transitions) == 0 {
+		// maxCodeSizeConfig not used. return
+		return nil, big.NewInt(0), big.NewInt(0)
+	}
+
+	// existing config had Transitions and new one does not have the same return error
+	if len(c1.Transitions) > 0 && len(c2.Transitions) == 0 {
+		return fmt.Errorf("genesis file missing transitions information"), head, head
+	}
+
+	if len(c2.Transitions) > 0 && len(c1.Transitions) == 0 {
+		return nil, big.NewInt(0), big.NewInt(0)
+	}
+
+	// check the number of records below current head in both configs
+	// if they do not match throw an error
+	c1RecsBelowHead := 0
+	for _, data := range c1.Transitions {
+		if data.Block.Cmp(head) <= 0 {
+			c1RecsBelowHead++
+		} else {
+			break
+		}
+	}
+
+	c2RecsBelowHead := 0
+	for _, data := range c2.Transitions {
+		if data.Block.Cmp(head) <= 0 {
+			c2RecsBelowHead++
+		} else {
+			break
+		}
+	}
+
+	// if the count of past records is not matching return error
+	if c1RecsBelowHead != c2RecsBelowHead {
+		return errors.New("transitions data incompatible. updating transitions for past"), head, head
+	}
+
+	// validate that each past record is matching exactly. if not return error
+	for i := 0; i < c1RecsBelowHead; i++ {
+		isSameBlock := c1.Transitions[i].Block.Cmp(c2.Transitions[i].Block) != 0
```

Comment:
if it is not the same block, that means some block(s) has(ve) disappear?
should we raise an error?